sale_block_no_stock/models/sale_order.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        if self.env.context.get("skip_block_no_stock_check"):
+            return super().action_confirm()
```

Comment:
issue(security): Unauthorized users shouldn't be able to forge an RPC request that includes this context to bypass the blocking. Please check user permissions before taking the context into account.